### ISSUE TITLE
Dealing with extended MIME types

### ISSUE THREAD

**Issue by KiChjang** [hyperium/mime#22]
When implementing the [fetch spec](https://fetch.spec.whatwg.org/#concept-fetch) for [servo](https://github.com/servo/servo), I came across MIME types such as `image/svg+xml` and `application/xhtml+xml`. Currently `mime.rs` does not contain any sublevel types that match these extended MIME types, and simply using, as in the two example, `Sublevel::Xml`, to represent them is not ideal, since these could be in a vector of MIME types in the header that may contain a `TopLevel` that matches and an actual `SubLevel` of `Xml`.

**Comment by seanmonstar** [comment]
For now, it's possible to use `SubLevel::Ext("svg+xml")`, I believe. But this would be a good thing to clean up. I wish the spec were more clear on **why** and **when** you'd use these extensions, but oh well.

**Comment by sanmai-NL** [comment]
@seanmonstar: In 0.2.2 [`SubLevel`](https://docs.rs/mime/0.2.2/mime/enum.SubLevel.html) isn't `Copy`, I presume because of the `Ext` variant that has a `String` field. In [`media-types`](https://docs.rs/media-types/0.2.0/media_types/enum.Type.html) this seems to have been solved better.

**Comment by seanmonstar** [comment]
Doesn't look like it's Copy there either.

**Comment by seanmonstar** [comment]
This has been fixed in 0.3.

```rust
let svg  = "image/svg+xml".parse().unwrap();
assert_eq!(svg.suffix(), Some(mime::XML));
```
